### DISCUSSION TITLE
Using predict() with regression variables.

### DISCUSSION THREAD

**Discussion by pascalschulthess** [nlmixr2/nlmixr2#248] [Q&A] [answered]
I want to simulate population predictions using `predict()`. Mainly I want to do this to have denser time steps of the prediction. I've tried
```
predict(fit, newdata = et(seq(0, 168, 1)))
```
This doesn't work because I have regression variables that I pull from the dataset and the newdata above of course doesn't know them.
So, how could I approach this?

**Comment by mattfidler** [comment, marked as answer]
Hi @pascalschulthess 

You would simply add the regressor variable to the datset.  

For instance,

```r
nd <- et(0:168) %>% et(id=1:10)
wt <- data.frame(id=1:10, WT=70*rnorm(10, sd=7))
nd <- merge(wt, nd)
predict(fit, newdata = nd)
```
